visualization/app/codeCharta/codeCharta.mouseEvent.service.ts

```diff
@@ -0,0 +1,31 @@
+import { StoreService } from "./state/store.service"
+import { PanelSelection, SearchPanelMode } from "./codeCharta.model"
+import { setPanelSelection } from "./state/store/appSettings/panelSelection/panelSelection.actions"
+import { setSearchPanelMode } from "./state/store/appSettings/searchPanelMode/searchPanelMode.actions"
+
+export class CodeChartaMouseEventService {
+	constructor(private storeService: StoreService) {}
+
+	public closeComponentExcept(fun?: Function) {
```

Comment:
Should this not always be called with an actual function?

Reply:
No. This function is triggered with no function, when clicking on the codeMap now. In this case, you want both functions in the array to be called in order to close all panels.